### PR TITLE
Add resource manager tags to GKE autopilot

### PR DESCRIPTION
**Pull request by maxi-cit**
Adds support for Resource Manager Tags for Autopilot GKE cluster resource

Part of: https://github.com/hashicorp/terraform-provider-google/issues/16614

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `node_pool_auto_config.resource_manager_tags` field to `google_container_cluster` resource
```

### CODE REVIEW

**Comment by SarahFrench** [comment]
Here's the feedback from the build failure above:

```
google-beta/services/container/resource_container_cluster_test.go:9683:5: undefined: serviceAccount
google-beta/services/container/resource_container_cluster_test.go:9697:5: undefined: serviceAccount
google-beta/services/container/resource_container_cluster_test.go:9705:6: undefined: serviceAccount
google-beta/services/container/resource_container_cluster_test.go:9760:47: undefined: enabled
google-beta/services/container/resource_container_cluster_test.go:9761:5: undefined: withNetworkTag
```

**Comment by maxi-cit** [comment]
Hello @SarahFrench, I updated this PR with fixes for the integration tests. I think I am ready for another round of review.
Could you run the tests please?

**Comment by maxi-cit** [comment]
Hello @SarahFrench could u tell me if there is something missing to merge this PR?

**Review by SarahFrench — COMMENTED** [review]
This PR looks good given the tests for the new field. There's a small refactor that we'd need before merging, to make sure the new test doesn't impact other acceptance tests.

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -9679,3 +9735,415 @@ resource "google_container_cluster" "primary" {
...
+
+resource "google_project_iam_binding" "tagHoldAdmin" {
+  project = "%[1]s"
+  role    = "roles/resourcemanager.tagHoldAdmin"
+  members = [
+   "serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com",
+  ]
+}
+
+resource "google_project_iam_binding" "tagUser" {
```
[the remark this answers is not in the document]
Comment (maxi-cit) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb]: updated

**Comment by maxi-cit** [comment]
Hello @SarahFrench, I updated the PR considering your suggestions, could you run the tests again please?

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 2]
```diff
@@ -9735,3 +9791,415 @@ resource "google_container_cluster" "primary" {
...
+data "google_project" "project" {
+  project_id = "%[1]s"
+}
+
+resource "google_project_iam_member" "tagHoldAdmin" {
+  project = "%[1]s"
+  role    = "roles/resourcemanager.tagHoldAdmin"
+  members = [
+   "serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com",
+  ]
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9806]:
```suggestion
  member = "serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com"
```

IAM membership resources manage only one membership in a binding, so cannot take a list as an argument here.

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 3]
```diff
@@ -9735,3 +9791,415 @@ resource "google_container_cluster" "primary" {
...
+}
+
+resource "google_project_iam_member" "tagUser" {
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:9948]: I'm afraid there'll need to be two membership resources used here to replace the original binding resource; _iam_membership resources only take a singular member argument

**Review by SarahFrench — COMMENTED** [review]
Thanks for doing the refactoring I suggested- I noticed some problems with use of the new _iam_membership resources and I left some comments. Those comments may need to be applied in other places where those resources are used.

**Comment by maxi-cit** [comment]
Hello @SarahFrench can I just roll back to google_project_iam_binding? I ve got the strong feeling I will mess up with the replaying mode.
Besides I already got a similar PR merged using iam binding resource
Check [this](https://github.com/GoogleCloudPlatform/magic-modules/pull/9531/files)

**Comment by SarahFrench** [comment]
Ok, roll back and I can update your PR to use _iam_member resources after it's merged. Even though you got a previous PR accepted while using _iam_binding resources that doesn't mean that it's ideal. Once your new test joins our full test suite it will have the potential to conflict with other tests that also try to create bindings for that role at the project level. This type of conflict is what the _iam_member resources are designed to avoid.

**Review by SarahFrench — APPROVED** [review]

**Pull request merged**